addons/website_sale/data/product_snippet_template_data.xml

```diff
@@ -0,0 +1,99 @@
+                <div class="o_carousel_product_card card h-100" t-att-data-url="record.website_url">
+                    <div t-if="is_sample" class="h5 o_ribbon_right bg-primary text-uppercase">Sample</div>
+                    <a class="o_carousel_product_img_link" t-att-href="record.website_url">
+                        <img class="card-img-top p-3" loading="lazy" t-att-src="data.get('image_512')" t-att-alt="record.display_name"/>
+                    </a>
+                    <div class="o_carousel_product_card_body card-body">
+                        <div class="card-title h5" t-field="record.display_name"/>
+                        <div class="card-text" t-raw="data.get('description_sale')"/>
+                        <t t-if="data.get('rating')" t-call="portal_rating.rating_widget_stars_static">
+                            <t t-set="rating_avg" t-value="record.rating_avg"/>
+                            <t t-set="rating_count" t-value="record.rating_count"/>
+                        </t>
+                    </div>
+                    <div class="o_carousel_product_card_footer card-footer d-flex align-items-center">
+                        <div class="card-text text-center" t-raw="data.get('price')"/>
+                        <a class="btn btn-primary ml-auto" t-att-href="record.website_url">
+                            <i class="fa fa-fw fa-eye"/>
+                        </a>
+                    </div>
+                </div>
+            </t>
+        </template>
+
+        <template id="dynamic_filter_template_mini_product_image" name="Image only">
+            <t t-foreach="records" t-as="data">
+                <t t-set="record" t-value="data.get('_record')"/>
+                <div class="card h-100" t-att-data-url="record.website_url">
+                    <div t-if="is_sample" class="h5 o_ribbon_right bg-primary text-uppercase">Sample</div>
+                    <a class="o_carousel_product_img_link" t-att-href="record.website_url">
+                        <img class="card-img-top p-3" loading="lazy" t-att-src="data.get('image_512')" t-att-alt="record.display_name"/>
+                    </a>
+                </div>
+            </t>
+        </template>
+
+        <template id="dynamic_filter_template_mini_product_price" name="Image with price">
+            <t t-foreach="records" t-as="data">
+                <t t-set="record" t-value="data.get('_record')"/>
+                <div class="card h-100" t-att-data-url="record.website_url">
+                    <div t-if="is_sample" class="h5 o_ribbon_right bg-primary text-uppercase">Sample</div>
+                    <a class="o_carousel_product_img_link" t-att-href="record.website_url">
+                        <img class="card-img-top p-3" loading="lazy" t-att-src="data.get('image_512')" t-att-alt="record.display_name"/>
+                    </a>
+                    <div class="card-text text-center" t-raw="data.get('price')"/>
```

Comment:
data['price']

as explain, when you know that you have the info use [] instead of get...
And in this case price is an html pre-generated so we always got it.